rotkehlchen/data/importer.py

```diff
@@ -469,3 +480,269 @@ def import_cryptocom_csv(self, filepath: Path) -> Tuple[bool, str]:
+        return True, ''
+
+    def _consume_nexo(self, csv_row: Dict[str, Any]) -> None:
+        if 'rejected' not in csv_row['Description']:
+            timestamp = deserialize_timestamp_from_date(
+                date=csv_row['DateTime'],
+                formatstr='%Y-%m-%d %H:%M:%S',
+                location='NEXO',
+            )
+        else:
+            raise DeserializationError('Action not confirmed yet')
+
+        asset = symbol_to_asset_or_token(csv_row['Currency'])
+        amount = deserialize_asset_amount_force_positive(csv_row['Amount'])
+        entry_type = csv_row['Type']
+
+        if entry_type in ('Deposit', 'ExchangeDepositedOn'):
+            asset_movement = AssetMovement(
+                location=Location.NEXO,
+                category=AssetMovementCategory.DEPOSIT,
+                address=None,
+                transaction_id=None,
+                timestamp=timestamp,
+                asset=asset,
+                amount=amount,
+                fee=Fee(ZERO),
+                fee_asset=A_USD,
+                link='',
+            )
+            self.db.add_asset_movements([asset_movement])
+        elif entry_type in ('Withdrawal', 'WithdrawExchanged'):
+            asset_movement = AssetMovement(
+                location=Location.NEXO,
+                category=AssetMovementCategory.WITHDRAWAL,
+                address=None,
+                transaction_id=None,
+                timestamp=timestamp,
+                asset=asset,
+                amount=amount,
+                fee=Fee(ZERO),
+                fee_asset=A_USD,
+                link='',
+            )
+            self.db.add_asset_movements([asset_movement])
+        elif entry_type == 'Withdrawal Fee':
+            action = LedgerAction(
+                identifier=0,  # whatever is not used at insertion
+                timestamp=timestamp,
+                action_type=LedgerActionType.EXPENSE,
+                location=Location.NEXO,
+                amount=amount,
+                asset=asset,
+                rate=None,
+                rate_asset=None,
+                link=None,
+                notes=f'{entry_type} from Nexo',
+            )
+            self.db_ledger.add_ledger_action(action)
+        elif entry_type in ('Interest', 'Bonus'):
+            action = LedgerAction(
+                identifier=0,  # whatever is not used at insertion
+                timestamp=timestamp,
+                action_type=LedgerActionType.INCOME,
+                location=Location.NEXO,
+                amount=amount,
+                asset=asset,
+                rate=None,
+                rate_asset=None,
+                link=None,
+                notes=f'{entry_type} from Nexo',
+            )
+            self.db_ledger.add_ledger_action(action)
+        else:
+            raise UnsupportedNexoEntry(f'Unsuported entry {entry_type}. Data: {csv_row}')
+
+    def import_nexo_csv(self, filepath: Path) -> Tuple[bool, str]:
```

Comment:
All these import functions are exactly the same. There is a lot of copy paste here. Can you try to abstract them properly? Can be done (and probably should) in a different PR.

Reply:
Oh my plan is to do it in another PR https://github.com/rotki/rotki/issues/2886